ios/RNLinksdk.m

```diff
@@ -273,54 +239,433 @@ - (PLKConfiguration*)getLegacyLinkConfiguration:(NSDictionary*)configuration {
-     linkConfiguration.accountSubtypes = accountSubtypes;
-  }
-  if ([countryCodes count] > 0) {
-     linkConfiguration.countryCodes = countryCodes;
-  }
-  if ([language length] > 0) {
-     linkConfiguration.language = language;
+      linkConfiguration.accountSubtypes = [RNLinksdk accountSubtypesArrayFromAccountSubtypesDictionary:accountSubtypes];
   }
 
   return linkConfiguration;
 }
 
-@end
+#pragma mark - Bridging
+
++ (PLKEnvironment)environmentFromString:(NSString *)string {
+    if ([string isEqualToString:@"production"]) {
+        return PLKEnvironmentProduction;
+    }
+    
+    if ([string isEqualToString:@"sandbox"]) {
+        return PLKEnvironmentSandbox;
+    }
+    
+    if ([string isEqualToString:@"development"]) {
+        return PLKEnvironmentDevelopment;
+    }
+    
+    // Default to Development
+    NSLog(@"Unexpected environment string value: %@. Expected one of: production, sandbox, or development.", string);
+    return PLKEnvironmentDevelopment;
+}
+
++ (NSArray<NSNumber *> *)productsArrayFromProductsStringArray:(NSArray<NSString *> *)productsStringArray {
+    NSMutableArray<NSNumber *> *results = [NSMutableArray arrayWithCapacity:productsStringArray.count];
+    
+    for (NSString *productString in productsStringArray) {
+        NSNumber *product = [self productFromProductString:productString];
+        if (product) {
+            [results addObject:product];
+        }
+    }
+    
+    return [results copy];
+}
+
++ (NSNumber * __nullable)productFromProductString:(NSString *)productString {
+    NSDictionary *productStringMap = @{
+        @"auth": @(PLKProductAuth),
+        @"identity": @(PLKProductIdentity),
+        @"income": @(PLKProductIncome),
+        @"transactions": @(PLKProductTransactions),
+        @"assets": @(PLKProductAssets),
+        @"liabilities": @(PLKProductLiabilities),
+        @"investments": @(PLKProductInvestments),
+    };
+    return productStringMap[productString.lowercaseString];
+}
+
++ (NSArray<id<PLKAccountSubtype>> *)accountSubtypesArrayFromAccountSubtypesDictionary:(NSDictionary<NSString *, NSArray<NSString *> *> *)accountSubtypesDictionary {
+    __block NSMutableArray<id<PLKAccountSubtype>> *results = [NSMutableArray array];
+    
+    for (NSString *type in accountSubtypesDictionary.allKeys) {
+        NSArray<NSString *> *subtypes = accountSubtypesDictionary[type] ?: @[];
+        
+        for (NSString *subtype in subtypes) {
+            id<PLKAccountSubtype> accountSubtype = [self accountSubtypeFromTypeString:type subtypeString:subtype];
+            if (accountSubtype) {
+                [results addObject:accountSubtype];
+            }
+        }
+    }
+    
+    return [results copy];
+}
+
++ (id<PLKAccountSubtype> __nullable)accountSubtypeFromTypeString:(NSString *)typeString
+                                                   subtypeString:(NSString *)subtypeString {
+    if ([typeString.lowercaseString isEqualToString:@"other"]) {
```

Comment:
Two foods for thought (not necessary for this release):

1. If Objective C handles Strings the way Java/Kotlin do (creates a new String instance when being "modified"), we probably want to store `typeString.lowercaseString` into a val and then call that rather than repeatedly modify the String

2. Instead of a series of `if`s, it's probably worth creating a static map of 
{"credit" to [PLKAccountSubtypeCredit createWithValue:PLKAccountSubtypeValueCreditPaypal],
// ...
"savings" to [PLKAccountSubtypeDepository createWithValue:PLKAccountSubtypeValueDepositorySavings];
}
 since we expect to hit this fairly often (an item can contain multiple accounts). I personally think it's a little more readable (but I'm not sure what the map syntax is like on objective C so maybe it sucks)

Reply:
The static map approach is interesting, I'll dive into it